packages/react-components/react-alert/src/components/Alert/useAlert.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+import { Button } from '@fluentui/react-button';
+import { CheckmarkCircleFilled, DismissCircleFilled, InfoFilled, WarningFilled } from '@fluentui/react-icons';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+
+import type { AlertProps, AlertState } from './Alert.types';
+
+/**
+ * Create the state required to render Alert.
+ *
+ * The returned state can be modified with hooks such as useAlertStyles_unstable,
+ * before being passed to renderAlert_unstable.
+ *
+ * @param props - props from this instance of Alert
+ * @param ref - reference to root HTMLElement of Alert
+ */
+export const useAlert_unstable = (props: AlertProps, ref: React.Ref<HTMLElement>): AlertState => {
+  const { intent } = props;
+
+  /** Determine the icon to render based on the intent */
+  let defaultIcon;
+  switch (intent) {
+    case 'success':
+      defaultIcon = <CheckmarkCircleFilled />;
+      break;
+    case 'error':
+      defaultIcon = <DismissCircleFilled />;
+      break;
+    case 'warning':
+      defaultIcon = <WarningFilled />;
+      break;
+    case 'info':
+      defaultIcon = <InfoFilled />;
+      break;
+  }
+
+  const icon = resolveShorthand(props.icon, {
+    defaultProps: {
+      children: defaultIcon,
+    },
+    required: !!(props.icon || props.intent),
```

Comment:
If there will be `props.icon` the slot will be rendered. I would simply do this:

```suggestion
    required: !!props.intent,
```

I haven't verified it, but it should work 😉 

```
```

Reply:
It works!